Bug 1417857 Added more into the azure docs

bfallonf opened the pull request:
As per: https://bugzilla.redhat.com/show_bug.cgi?id=1417857

More work to come.

tanaka-takayoshi commented:
Hi @bfallonf  , could you fix the indent of ocp 3.5 release notes?
https://github.com/openshift/openshift-docs/blob/master/release_notes/ocp_3_5_release_notes.adoc#dynamic-provisioner-for-azure-block-storage

bfallonf commented:
I've updated the rhel notes directly in the github branch.

But I've also updated the docs as per this BZ and [BZ1422109](https://bugzilla.redhat.com/show_bug.cgi?id=1422109).

I'll put the BZ on QA and see if there's any feedback.

rootfs commented on install_config/configuring_azure.adoc:
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
+If the file does not exist, create it, and add the following:
+
+----
+tenantId: <> <1>
```
This file is in json format. Can you reformat it?

weherdh replied:
Json format is like below:
```
{
  "aadClientID" : ...,
  "aadClientSecret" : ...,
  "subscriptionID" : ...,
  "tenantID" : ...,
  "resourceGroup": ...,
}
```

rootfs commented on install_config/configuring_azure.adoc:
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
+<2> The Azure subscription ID that the cluster is deployed in.
+<3> The client ID for an AAD application with RBAC access to talk to Azure RM APIs.
+<4> The client secret for an AAD application with RBAC access to talk to Azure RM APIs.
+<5> Ensure this is the same as tenant ID.
```
it is optional

rootfs commented on install_config/configuring_azure.adoc:
```diff
@@ -19,6 +19,30 @@ Azure disk as persistent storage] for application data. After Azure is
+<4> The client secret for an AAD application with RBAC access to talk to Azure RM APIs.
+<5> Ensure this is the same as tenant ID.
+<6> The Azure Resource Group name that Azure VM belongs to.
+<7> The compact style Azure region, for example `southeastasia`.
```
it is optional

bfallonf commented:
@rootfs Thanks for the review. I'm not sure why I need to reformat the file example. I think it is already in YAML format. Are you saying it should be in JSON format? I've tried to use a formatter to change it, but it's giving errors.

I've also updated the examples to say the two are optional.

If there's anything else, please let me know.

bfallonf commented:
Confirmed that YAML is fine for the file layout.

@adellape @ahardin-rh @bmcelvee @mburke5678 @gaurav-nelson peer review plz?

bfallonf commented:
For reference, I added information about the azure.conf file and added a prerequisites section.

rootfs commented:
lgtm

ahardin-rh commented:
LGTM! :fireworks:

bfallonf commented:
Thanks @ahardin-rh . I'll merge.

bfallonf commented:
[rev_history]
|xref:../install_config/configuring_azure.adoc#install-config-configuring-azure[Configuring For Azure]
|Added the xref:../install_config/configuring_azure.adoc#azure-configuration-file[The Azure Configuration File] section and some file example fixes.
%
|xref:../install_config/persistent_storage/persistent_storage_azure.adoc#install-config-persistent-storage-persistent-storage-azure[Configuring Persistent Storage -> Persistent Storage Using Azure Disk]
|Added the xref:../install_config/persistent_storage/persistent_storage_azure.adoc#azure-prerequisites[Prerequisites] section.
%
